llvm/lib/Target/RISCV/RISCVAsmPrinter.cpp

```diff
@@ -290,6 +293,108 @@ bool RISCVAsmPrinter::EmitToStreamer(MCStreamer &S, const MCInst &Inst,
 // instructions) auto-generated.
 #include "RISCVGenMCPseudoLowering.inc"
 
+void RISCVAsmPrinter::emitGlobalVariable(const GlobalVariable *GV) {
+  auto CheriotCapImportAttrName =
+      llvm::cheri::CapabilityImportAttr::getAttrName();
+
+  if (GV->hasAttribute(CheriotCapImportAttrName)) {
+    // This global variable refers to an object exported from a different
+    // compartment.
+    //
+    // The final result should be that of having a section in the object file
+    // that, given a <name> for the object to import, introduces a comdat (i.e.
+    // can appear multiple times across different objects without causing linker
+    // errors) symbol in `.compartment_imports.<name>`.
+    //
+    // This symbol defines a capability that can be used to interact with the
+    // object exported from the other compartment.
+    //
+
+    if (GV->isLocalLinkage(GV->getLinkage())) {
```

Comment:
This can be an assert, the back end is allowed to assume that IR is well formed. Maybe move this check to the Verifier?

Reply:
cc8227d02ff6bea3bdab662e5d7853b3e6d61a97 moves that check in `llvm/lib/IR/Verifier.cpp`, and replaces the quotation above with an assertion.